src/plugins/data_views/common/data_views/data_view_lazy.ts

```diff
@@ -0,0 +1,619 @@
+      fld.spec.esTypes = [fieldType];
+    } else {
+      fld = new DataViewField({
+        name: fieldName,
+        runtimeField: runtimeFieldSpec,
+        type: castEsToKbnFieldTypeName(fieldType),
+        esTypes: [fieldType],
+        aggregatable: true,
+        searchable: true,
+        count: config.popularity ?? 0,
+        readFromDocValues: false,
+        shortDotsEnable: this.shortDotsEnable,
+      });
+      this.fieldCache.set(fieldName, fld);
+    }
+
+    this.setFieldCustomLabel(fieldName, config.customLabel);
+    this.setFieldCustomDescription(fieldName, config.customDescription);
+
+    if (config.popularity || config.popularity === null) {
+      this.setFieldCount(fieldName, config.popularity);
+    }
+
+    if (config.format) {
+      this.setFieldFormat(fieldName, config.format);
+    } else if (config.format === null) {
+      this.deleteFieldFormat(fieldName);
+    }
+
+    return fld;
+  }
+
+  private getRuntimeFieldSpecMap = ({ fieldName = ['*'] }: Pick<GetFieldsParams, 'fieldName'>) => {
+    const spec: DataViewFieldSpecMap = {};
+
+    const addRuntimeFieldToSpecFields = (
+      name: string,
+      fieldType: RuntimeType,
+      runtimeField: RuntimeFieldSpec,
+      parentName?: string
+    ) => {
+      spec[name] = {
+        name,
+        type: castEsToKbnFieldTypeName(fieldType),
+        esTypes: [fieldType],
+        runtimeField,
+        aggregatable: true,
+        searchable: true,
+        readFromDocValues: false,
+        customLabel: this.fieldAttrs?.[name]?.customLabel,
+        customDescription: this.fieldAttrs?.[name]?.customDescription,
+        count: this.fieldAttrs?.[name]?.count,
+      };
+
+      if (parentName) {
+        spec[name].parentName = parentName;
+      }
+    };
+
+    // CREATE RUNTIME FIELDS
+    for (const [name, runtimeField] of Object.entries(this.runtimeFieldMap || {})) {
+      if (fieldMatchesFieldsRequested(name, fieldName)) {
+        // For composite runtime field we add the subFields, **not** the composite
+        if (runtimeField.type === 'composite') {
+          Object.entries(runtimeField.fields!).forEach(([subFieldName, subField]) => {
+            addRuntimeFieldToSpecFields(
+              `${name}.${subFieldName}`,
+              subField.type,
+              runtimeField,
+              name
+            );
+          });
+        } else {
+          addRuntimeFieldToSpecFields(name, runtimeField.type, runtimeField);
+        }
+      }
+    }
+
+    return spec;
+  };
+
+  private getScriptedFields({ fieldName = ['*'] }: Pick<GetFieldsParams, 'fieldName'>) {
+    const dataViewFields: Record<string, DataViewField> = {};
+
+    Object.values(this.scriptedFieldsMap).forEach((field) => {
+      if (!fieldMatchesFieldsRequested(field.name, fieldName)) {
+        return;
+      }
+      let fld = this.fieldCache.get(field.name);
+
+      if (fld) {
+        this.fieldCache.delete(field.name);
+      }
```

Comment:
Also curious why we delete and re-add scripted fields to the cache when we get them?

Reply:
Scripted fields override mapped fields. Added a comment to the code.